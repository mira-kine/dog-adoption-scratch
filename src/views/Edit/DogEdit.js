import React from 'react';
import { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom/cjs/react-router-dom.min';
import DogForm from '../../components/Form/DogForm';
import Header from '../../components/Header/Header';
import { fetchDogById, updateDog } from '../../services/dogs';

export default function DogEdit() {
  const [selectedDog, setDog] = useState({});
  const [loading, setLoading] = useState(true);
  const params = useParams();

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogById(params.id);
      setDog(data);
      setLoading(false);
    };
    fetchData();
  }, [params.id]);

  if (loading) {
    return <h1>loading</h1>;
  }

  const updateDogValue = (key, value) => {
    selectedDog[key] = value;
    setDog({ ...selectedDog });
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    await updateDog(selectedDog);
    alert("You've successfully updated your dog");
  };

  return (
    <div>
      <Header />
      <p>
        <DogForm
          selectedDog={{ ...selectedDog }}
          handleSubmit={handleSubmit}
          updateDogValue={updateDogValue}
        />
      </p>
    </div>
  );
}
