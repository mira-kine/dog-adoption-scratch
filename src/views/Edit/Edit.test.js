import { MemoryRouter, Route } from 'react-router-dom/cjs/react-router-dom.min';
import DogEdit from '../Edit/DogEdit';
import { render, screen } from '@testing-library/react';

test('should render an individual dog', async () => {
  // render the component
  const { container } = render(
    <MemoryRouter initialEntries={['/dogs/6/edit']}>
      <Route path="/dogs/:id/edit" component={DogEdit} />
    </MemoryRouter>
  );
  await screen.findByDisplayValue('Mookie');
  expect(container).toMatchSnapshot();
});
