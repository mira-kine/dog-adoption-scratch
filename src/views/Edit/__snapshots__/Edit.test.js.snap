// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an individual dog 1`] = `
<div>
  <div>
    <div>
      <a
        href="/"
      >
        Home
      </a>
    </div>
    <div
      class="form-container"
    >
      <form
        class="dog-form"
      >
        <h1>
          Dog Create/Edit Form
        </h1>
        <label>
          Name:
        </label>
        <input
          name="name"
          placeholder="name"
          type="text"
          value="Mookie"
        />
        <label>
          Age:
        </label>
        <input
          name="age"
          placeholder="age"
          type="text"
          value="1"
        />
        <label>
          Breed:
        </label>
        <input
          name="breed"
          placeholder="breed"
          type="text"
          value="Goldendoodle"
        />
        <label>
          Image:
        </label>
        <input
          name="image"
          placeholder="image"
          type="text"
          value="https://placedog.net/500?id=200
"
        />
        <label>
          Bio:
        </label>
        <input
          name="bio"
          placeholder="bio"
          type="text"
          value="Mellow and full of energy at the same time. Hypoallergenic."
        />
        <button
          class="submit-button"
        >
          Add Dog
        </button>
      </form>
    </div>
  </div>
</div>
`;
